src/comments.js

```diff
@@ -153,8 +154,12 @@ function attach(comments, ast, text) {
       // We also need to check if it's the first line of the file.
 
       if (followingNode) {
-        // Always a leading comment.
-        addLeadingComment(followingNode, comment);
+        // There are some exceptions like in-between comments.
+        if (enclosingNode && enclosingNode.type === 'IfStatement') {
```

Comment:
Hm, this comment code is already very complicated so I'd love to find a way to that doesn't special-case `if`s in here. This code is going to become unmaintainable if we allow special-cases; even without them it's already very hard to understand.